cli/debug.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+)
+
+// writeTerragruntDebugFile will create a tfvars file that can be used to invoke the terraform module in the same way
+// that terragrunt invokes the module, so that you can debug issues with the terragrunt config.
+func writeTerragruntDebugFile(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
+	terragruntOptions.Logger.Printf("Debug mode requested: generating debug file")
+
+	variables, err := terraformModuleVariables(terragruntOptions)
+	if err != nil {
+		return err
+	}
+	util.Debugf(terragruntOptions.Logger, "The following variables were detected in the terraform module:")
+	util.Debugf(terragruntOptions.Logger, "%v", variables)
+
+	fileContents, err := terragruntDebugFileContents(terragruntOptions, terragruntConfig, variables)
+	if err != nil {
+		return err
+	}
+
+	configFolder := filepath.Dir(terragruntOptions.TerragruntConfigPath)
+	fileName := filepath.Join(configFolder, "terragrunt-debug.tfvars.json")
+	if err := ioutil.WriteFile(fileName, fileContents, os.FileMode(int(0600))); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	terragruntOptions.Logger.Printf("Variables passed to terraform are located in \"%s\"", fileName)
+	terragruntOptions.Logger.Printf("Run this command to replicate how terraform was invoked:")
+	terragruntOptions.Logger.Printf("\tterraform apply -var-file=\"%s\" \"%s\"", fileName, terragruntOptions.WorkingDir)
+	return nil
+}
+
+// terragruntDebugFileContents will return a tfvars file in json format of all the terragrunt rendered variables values
+// that should be set to invoke the terraform module in the same way as terragrunt. Note that this will only include the
+// values of variables that are actually defined in the module.
+func terragruntDebugFileContents(
+	terragruntOptions *options.TerragruntOptions,
+	terragruntConfig *config.TerragruntConfig,
+	moduleVariables []string,
+) ([]byte, error) {
+	var envVars map[string]string
+	if terragruntOptions.Env == nil {
+		envVars = map[string]string{}
+	} else {
+		envVars = terragruntOptions.Env
+	}
```

Comment:
NIT: slightly shorter.

```suggestion
	envVars := map[string]string{}
	if terragruntOptions.Env != nil {
		envVars = terragruntOptions.Env
	}
```

Reply:
Done: acf43b5